lib/sda/service.go

```diff
@@ -278,3 +282,120 @@ func CreateServiceMessage(service string, r interface{}) (*ServiceMessage, error
 	}, nil
 
 }
+
+/*
+A simple client structure to be used when wanting to connect to services. It
+holds the private and public key and allows to connect to a service through
+the network.
+The error-handling is done using the ErrorRet structure which can be returned
+in place of the standard reply. The Client.Send method will catch that and return
+ the appropriate error.
+*/
+
+// Client for a service
+type Client struct {
+	Private abstract.Secret
```

Comment:
I would put `Private` to `private`, or even not keeping at as a field. But here, it says it may be useful to a user to play with that `Private` while it's not.